biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java

```diff
@@ -127,6 +128,9 @@
 
     protected static final Pattern readableFiles = Pattern.compile(".*(g[bp]k*$|\\u002eg[bp].*)");
     protected static final Pattern headerLine = Pattern.compile("^LOCUS.*");
+	private static final String DBSOURCE = "DBSOURCE";

```

Comment:
HI Paolo,
when I include the little print function in GenbankSequenceParser:
    private void printSection(List<String[]> sec) {
        if(sec!=null && sec.size()>0) {
            for(String[] sa : sec) {
                StringBuffer sb=new StringBuffer();
                for(String s : sa) {
                    sb.append(" "+s);
                }
                log.debug(new String(sb)+"\n");
            }
        }
    }

and let it run for the SOURCE and DBSOURCE tags I get quite different results:
for Example during the test case Reading: /tmp/254839678.gb
 DBSOURCE pdb: molecule 3IAN, chain 65, release Jul 29, 2009;
deposition: Jul 14, 2009;
class: Hydrolase;
source: Mmdb_id: 75718, Pdb_id 1: 3IAN;
Exp. method: X-Ray Diffraction.
 SOURCE Lactococcus lactis subsp. lactis
 ORGANISM Lactococcus lactis subsp. lactis
Bacteria; Firmicutes; Bacilli; Lactobacillales; Streptococcaceae;
Lactococcus.

while for test case Reading: /tmp/NP_000257.gb
 DBSOURCE REFSEQ: accession NM_000266.3
 SOURCE Homo sapiens (human)
 ORGANISM Homo sapiens
Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;
Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini;
Catarrhini; Hominidae; Homo.

I would conclude, that the two section keys are not redundant
CheersStefan



     paolopavan <notifications@github.com> schrieb am 23:29 Donnerstag, 12.Februar 2015:
   

 In biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java:> @@ -127,6 +128,9 @@
>  
>      protected static final Pattern readableFiles = Pattern.compile(".*(g[bp]k*$|\\u002eg[bp].*)");
>      protected static final Pattern headerLine = Pattern.compile("^LOCUS.*");
> +	private static final String DBSOURCE = "DBSOURCE";
Isn't SOURCE_TAG redundant to your DBSOURCE ?—
Reply to this email directly or view it on GitHub.